src/internet_identity/tests/integration/vc_mvp.rs

```diff
@@ -0,0 +1,805 @@
+
+    if let GetIdAliasResponse::NoSuchCredentials(_err) = response {
+        Ok(())
+    } else {
+        panic!("Expected that credentials not found, got {:?}", response);
+    }
+}
+
+/// Verifies that there is a graceful failure if II gets upgraded between prepare_id_alias
+/// and get_id_alias.
+#[test]
+fn should_not_get_prepared_id_alias_after_ii_upgrade() -> Result<(), CallError> {
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let identity_number = flows::register_anchor(&env, canister_id);
+    let relying_party = FrontendHostname::from("https://some-dapp.com");
+    let issuer = FrontendHostname::from("https://some-issuer.com");
+    let prepare_id_alias_req = PrepareIdAliasRequest {
+        identity_number,
+        relying_party: relying_party.clone(),
+        issuer: issuer.clone(),
+    };
+
+    let prepare_response = api::vc_mvp::prepare_id_alias(
+        &env,
+        canister_id,
+        principal_1(),
+        prepare_id_alias_req.clone(),
+    )?
+    .expect("Got 'None' from prepare_id_alias");
+
+    let prepared_id_alias = if let PrepareIdAliasResponse::Ok(prepared_id_alias) = prepare_response
+    {
+        prepared_id_alias
+    } else {
+        panic!("prepare id_alias failed")
+    };
+
+    // upgrade, even with the same WASM clears non-stable memory
+    upgrade_ii_canister(&env, canister_id, II_WASM.clone());
+
+    let get_id_alias_req = GetIdAliasRequest {
+        identity_number,
+        relying_party,
+        issuer,
+        rp_id_alias_jwt: prepared_id_alias.rp_id_alias_jwt,
+        issuer_id_alias_jwt: prepared_id_alias.issuer_id_alias_jwt,
+    };
+    let response = api::vc_mvp::get_id_alias(&env, canister_id, principal_1(), get_id_alias_req)?
+        .expect("Got 'None' from get_id_alias");
+
+    if let GetIdAliasResponse::NoSuchCredentials(_err) = response {
+        Ok(())
+    } else {
+        panic!("Expected that credentials not found, got {:?}", response);
+    }
+}
+
+#[test]
+#[should_panic(expected = "id_alias signature invalid")]
+fn should_not_validate_id_alias_with_wrong_canister_key() {
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let identity_number = flows::register_anchor(&env, canister_id);
+    let relying_party = FrontendHostname::from("https://some-dapp.com");
+    let issuer = FrontendHostname::from("https://some-issuer.com");
+    let prepare_id_alias_req = PrepareIdAliasRequest {
+        identity_number,
+        relying_party: relying_party.clone(),
+        issuer: issuer.clone(),
+    };
+
+    let prepare_response = api::vc_mvp::prepare_id_alias(
+        &env,
+        canister_id,
+        principal_1(),
+        prepare_id_alias_req.clone(),
+    )
+    .expect("Result of prepare_id_alias is not Ok")
+    .expect("Got 'None' from prepare_id_alias");
+
+    let prepared_id_alias = if let PrepareIdAliasResponse::Ok(prepared_id_alias) = prepare_response
```

Comment:
You could use the `match_value!` macro here to simplify things.

Reply:
Hmm, but `match_value!` does not give me the matched value back, i.e. I'd still need to match manually to get `prepared_id_alias`, no?

Reply:
`match_value!` supports bindings (see [here](https://github.com/dfinity/internet-identity/blob/49f0d0b721815c244bb0dc6c0e4c9514824ba3b2/src/internet_identity/tests/integration/v2_api/authn_method_add.rs#L38) for an example).

Reply:
TIL ^^, thanks for the example, done.